asserts/fskeypairmgr.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path/filepath"
+)
+
+// the default simple filesystem based keypair manager/backstore
+
+const (
+	privateKeysLayoutVersion = "v0"
+	privateKeysRoot          = "private-keys-" + privateKeysLayoutVersion
+)
+
+type filesystemKeypairManager struct {
+	top string
+}
+
+func newFilesystemKeypairMananager(path string) *filesystemKeypairManager {
```

Comment:
@mvo5 yes we have very similar tests going through the database that covers it, if we make the constructor exported it may be worth to have some direct tests though